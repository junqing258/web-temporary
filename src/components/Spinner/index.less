.bc_spinner--warp {
  position: fixed;
  z-index: 9;
  width: 100%;
  height: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.7);
  display: flex;
  justify-content: center;
  align-items: center;
}
