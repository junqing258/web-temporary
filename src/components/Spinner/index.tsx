import SVG from 'react-inlinesvg';

import './index.less';

const Loading: React.FC = () => {
  return (
    <div className="bc_spinner--warp">
      <SVG src={require('@/assets/loading-spin.svg').default} />
    </div>
  );
};
export default Loading;
