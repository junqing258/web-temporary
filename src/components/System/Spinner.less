@color: #55657e;
@bg: #0f151f;

.loader-warper {
  height: calc(100vh - 70px);
}

.loader {
  font-size: 10px;
  margin: 20px auto;
  text-indent: -9999em;
  width: 3em;
  height: 3em;
  border-radius: 50%;
  background: @color;
  background: linear-gradient(to right, @color 10%, rgba(255, 255, 255, 0) 42%);
  position: relative;
  animation: load3 0.8s infinite linear;
  transform: translateZ(0);

  position: fixed;
  z-index: 3;
  left: 50%;
  margin-left: -1.5em;
  top: 70px;
}
.loader:before {
  width: 50%;
  height: 50%;
  background: @color;
  border-radius: 100% 0 0 0;
  position: absolute;
  top: 0;
  left: 0;
  content: '';
}
.loader:after {
  background: @bg;
  width: 75%;
  height: 75%;
  border-radius: 50%;
  content: '';
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}
@-webkit-keyframes load3 {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
@keyframes load3 {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
