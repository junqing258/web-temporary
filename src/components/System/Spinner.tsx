import React from 'react';

import './Spinner.less';

const Spinner: React.FC = () => {
  return (
    <div className="loader-warper">
      <div className="loader">Loading...</div>
    </div>
  );
};

export default Spinner;
