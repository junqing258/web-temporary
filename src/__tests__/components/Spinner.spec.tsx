import React from 'react';
import { render, screen } from '@testing-library/react';
import Spinner from '@/components/Spinner';

test('text value', () => {
  const a = 1;
  expect(a === 1).toBe(true);
});

test('Basic text value', () => {
  const { container, asFragment } = render(
    <Spinner>
      <p>hello</p>
    </Spinner>,
  );
  expect(container.firstChild).toHaveClass('bc_spinner--warp');
  expect(asFragment()).toMatchSnapshot();
});
