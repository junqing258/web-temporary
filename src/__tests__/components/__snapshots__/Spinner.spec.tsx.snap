// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic text value 1`] = `
<DocumentFragment>
  <div
    class="bc_spinner--warp"
  />
</DocumentFragment>
`;
